src/kOS/Module/kOSLightModule.cs

```diff
@@ -28,23 +28,24 @@ internal class kOSLightModule : PartModule
 
         [KSPField(isPersistant = false, guiActive = false)]
         public string animationName = "";
-        
+
         [KSPField(isPersistant = false, guiActive = false)]
         public string pulseWidth = "";
 
         [KSPField(isPersistant = false, guiActive = false)]
         public string gapWidth = "";
 
         private double lastTime = double.MaxValue;
+        private Color lastColor = new Color(0, 0, 0, 0);
 
         private readonly Color powerOffColor = new Color(0, 0, 0, 0);
 
         private ModuleLight lightModule;
-        private Light[] lights;
-        private Renderer[] renderers;
+        private List<Light> lights;
+        private List<Renderer> renderers;
```

Comment:
In this particular case, I see no benefit.  I don't know why Squad did it, but they did.